controllers/argocd/deployment.go

```diff
@@ -492,7 +502,9 @@ func (r *ReconcileArgoCD) reconcileRedisDeployment(cr *argoproj.ArgoCD, useTLS b
 					"ALL",
 				},
 			},
-			RunAsNonRoot: boolPtr(true),
+			ReadOnlyRootFilesystem: boolPtr(true),
+			RunAsNonRoot:           boolPtr(true),
+			RunAsUser:              int64Ptr(999),
```

Comment:
You can remove this change. We made a change related to UIDs in Redis in #1668. However looks like it didn't go as planned. I am discussing with @Mangaal on this.

Reply:
Without UID change, E2E tests fail as the Because the container image user that is being used, cannot determine if its root or not. This is for the container image for `argo-rollouts-extensions`

Reply:
Will remove for redis deployment.

Reply:
Yes. I meant for redis deployment.
The redis changes @Mangaal made are working as expected. It was a false alarm.